File: Task/ABC-words/Nim/abc-words.nim
import strutils

func isAbcWord(word: string): bool =
  let ia = word.find('a')
  if ia < 0: return false
  let ib = word.find('b')
  if ib < ia: return false
  let ic = word.find('c')
  if ic < ib: return false
  result = true

var count = 0
for word in "unixdict.txt".lines:
  if word.isAbcWord:
    inc count
    echo ($count).align(2), ' ', word

File: Task/ABC-words/Rust/abc-words.rs
use std::fs;

fn isabcword(word: &&str) -> bool {
    let positions = ['a', 'b', 'c']
        .iter()
        .filter_map(|c| word.find(*c))
        .collect::<Vec<usize>>();
    return positions.len() == 3 && positions.windows(2).all(|w| w[0] <= w[1]);
}

fn main() {
    let wordsfile = fs::read_to_string("unixdict.txt").unwrap().to_lowercase();
    let words = wordsfile.split_whitespace().filter(|w| isabcword(w));
    for (i, w) in words.enumerate() {
        print!("{:<14}{}", w, if (i + 1) % 6 == 0 { "\n" } else { "" });
    }
    return ();
}
